Jenkinsfile

```
@@ -75,7 +101,163 @@ void createPipelineTriggers() {
     }
 }
 
-def step_compile_results()
+/* Gets the robot framework url and commit id based on pipeline parameters.
+ *
+ * Sets rfUrl and rfCommitId.
+ */
+def stepGetRfUrlAndCommit() {
+    (rfUrl, rfCommitId) = getUrlAndCommitFromParams(
+            params.HIL_RF_VERSION, "RIOT-OS",
+            "RobotFW-Tests", params.HIL_RF_PULL)
+    sh script: "echo rfUrl ${rfUrl}", label: "rfUrl ${rfUrl}"
+    sh script: "echo rfCommitId ${rfCommitId}",
+            label: "rfUrl ${rfCommitId}"
+}
+
+/* Gets the riot url and commit id based on pipeline parameters.
+ *
+ * Sets riotUrl and riotCommitId.
+ */
+def stepGetRiotUrlAndCommit() {
+    (riotUrl, riotCommitId) = getUrlAndCommitFromParams(
+            params.HIL_RIOT_VERSION, "RIOT-OS",
+            "RIOT", params.HIL_RIOT_PULL)
+    sh script: "echo riotUrl ${riotUrl}", label: "riotUrl ${riotUrl}"
+    sh script: "echo riotCommitId ${riotCommitId}",
+            label: "riotCommitId ${riotCommitId}"
+}
+
+/* Gets github url and commit based on parameters.
+ *
+ * @param ver           Type of version to get {master, pull, current, submodule}
+ * @param repo_owner    Repoisitory owner
+ * @param repo_name     Repoisitory name
+ * @param pull          Optional pull request number
+ *
+ * @return (url, commit)
+ */
+def getUrlAndCommitFromParams(ver, repo_owner, repo_name, pull = 0) {
+    if (ver == 'master') {
+        return getUrlAndCommitFromDefault(repo_owner, repo_name)
+    }
+    else if (ver == 'pull') {
+        assert pull != '0' : "Selected pull of ${repo_name} but no pull entered"
+        return getUrlAndCommitFromPr(repo_owner, repo_name, pull)
+    }
+    else if (ver == 'current') {
+        return ["https://github.com/${repo_owner}/${repo_name}",
```

```suggestion
        return ["https://github.com/RIOT-OS/RobotFW-tests",
```

Might as well specify rf repo directly as we are reading `env.GIT_COMMIT` which is the current commit in rf repo. In 'submodule' we also use rf directly, so no problem this being too specific I guess.